stock_request_employee/models/stock_request_order.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+from odoo import api, fields, models
+
+
+class StockRequestOrder(models.Model):
+
+    _inherit = 'stock.request.order'
+
+    to_employee = fields.Boolean()
+    employee_id = fields.Many2one(
+        'hr.employee',
+    )
+
+    @api.onchange('employee_id')
+    def _onchange_employee(self):
+        self.change_childs()
+
+    @api.onchange('to_employee')
+    def _onchange_to_employee(self):
+        if self.to_employee:
+            self.location_id = self.env.ref(
+                'stock_request_employee.location_employee'
+            )
+            if self.env.user.employee_ids:
+                self.employee_id = self.env.user.employee_ids[0]
+        else:
+            self.employee_id = False
+            self.location_id = False
+        self.change_childs()
+
+    def change_childs(self):
+        res = super().change_childs()
+        if not self._context.get('no_change_childs', False):
+            for line in self.stock_request_ids:
+                line.employee_id = self.employee_id
+                line.to_employee = self.to_employee
+        return res
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        """Add to the existing context of the field `stock_request_ids`
```

Comment:
To avoid all this stuff, why not creating a new action with the good default_... context ? That could make sense.

Reply:
The problem is that we want to fix a one2many field that already has a context. We cannot update the field using the xml and your proposal would fix the `to_employee` field, but not the `employee_id`, as it can be changed.

We can make two menus, but this code cannot be totally removed. What do you prefer?